spotify-recent-plays/src/Connector.js

```diff
@@ -0,0 +1,218 @@
+/* istanbul ignore next */
+if (typeof(require) !== 'undefined') {
+  var DataBuilder = require('./DataBuilder.js')['default'];
+  var Oauth2Builder = require('./Oauth2Builder.js')['default'];
+  var OauthService = require('./OauthService.js')['default'];
+  var SpotifyClient = require('./SpotifyClient.js')['default'];
+}
+
+/**
+ * Constructor for Connector - a wrapper for GDS connector methods
+
+ * @param {object} services - an object containing required Google services
+ *
+ * @return {object} a Connector object.
+ */
+function Connector(services) {
+  this.services = services;
+
+  return this;
+}
+
+/**
+ * @return {object} Object containing field definitions
+ */
+Connector.prototype.getSchema = function() {
+  return {
+    schema: [
+      {
+        name: 'track_name',
+        label: 'Track Name',
+        dataType: 'STRING',
+        semantics: {
+          conceptType: 'DIMENSION'
+        }
+      },
+      {
+        name: 'artist',
+        label: 'Artist',
+        dataType: 'STRING',
+        semantics: {
+          conceptType: 'DIMENSION'
+        }
+      },
+      {
+        name: 'played_at_hour',
+        label: 'Played at (date + hour)',
+        dataType: 'STRING',
+        semantics: {
+          conceptType: 'DIMENSION',
+          semanticGroup: 'DATETIME',
+          semanticType: 'YEAR_MONTH_DAY_HOUR'
+        }
+      },
+      {
+        name: 'played_at_date',
+        label: 'Played at (date)',
+        dataType: 'STRING',
+        semantics: {
+          conceptType: 'DIMENSION',
+          semanticGroup: 'DATETIME',
+          semanticType: 'YEAR_MONTH_DAY'
+        }
+      },
+      {
+        name: 'plays',
+        label: 'Plays',
+        dataType: 'NUMBER',
+        formula: 'COUNT(track_name)',
+        semantics: {
+          conceptType: 'METRIC',
+          isReaggregatable: false
+        }
+      },
+      {
+        name: 'tracks_count',
+        label: 'Played Tracks',
+        dataType: 'NUMBER',
+        formula: 'COUNT(track_name)',
+        semantics: {
+          conceptType: 'METRIC',
+          isReaggregatable: false
+        }
+      },
+      {
+        name: 'popularity',
+        label: 'Popularity',
+        dataType: 'NUMBER',
+        semantics: {
+          conceptType: 'METRIC'
+        }
+      }
+    ]
+  };
+};
+
+/**
+ * @return {object} Object containing connector configuration
+ */
+Connector.prototype.getConfig = function() {
+  return {
+    dateRangeRequired: true
+  };
+};
+
+/**
+ * @return {object} Object containing auth config
+ */
+Connector.prototype.getAuthType = function() {
+  return { type: 'OAUTH2' };
+};
+
+/**
+ * @return {boolean}
+ */
+Connector.prototype.isAdminUser = function() {
```

Comment:
This should return false, or have logic that actually checks if the current user is the admin of the connector.